libvirt/resource_libvirt_network_test.go

```diff
@@ -48,6 +50,130 @@ func TestNetworkAutostart(t *testing.T) {
 	})
 }
 
+func TestNetworkDNS(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckLibvirtNetworkDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: fmt.Sprintf(`
+				resource "libvirt_network" "test_net" {
+					name      = "networktest"
+					mode      = "nat"
+					domain    = "k8s.local"
+					addresses = ["10.17.3.0/24"]
+					dns {
+						local_only = true
+						forwarders = [
+						  {
+						    address = "8.8.8.8",
+					          },
+						  {
+						    address = "10.10.0.67",
+						    domain = "my.domain.com",
+						  },
+						  {
+						    domain = "hello.com",
+						  },
+						]
+					}
+				}`),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("libvirt_network.test_net", "dns.0.local_only", "true"),
+					testAccCheckLibvirtNetworkLocalOnlyStatus("libvirt_network.test_net", true),
+					resource.TestCheckResourceAttr("libvirt_network.test_net", "dns.0.forwarders.#", "3"),
+					resource.TestCheckResourceAttr("libvirt_network.test_net", "dns.0.forwarders.0.address", "8.8.8.8"),
+					resource.TestCheckResourceAttr("libvirt_network.test_net", "dns.0.forwarders.1.address", "10.10.0.67"),
+					resource.TestCheckResourceAttr("libvirt_network.test_net", "dns.0.forwarders.1.domain", "my.domain.com"),
+					resource.TestCheckResourceAttr("libvirt_network.test_net", "dns.0.forwarders.2.domain", "hello.com"),
+					testAccCheckLibvirtNetworkDNSForwarders("libvirt_network.test_net", []libvirtxml.NetworkDNSForwarder{
+						{
+							Addr: "8.8.8.8",
+						},
+						{
+							Addr:   "10.10.0.67",
+							Domain: "my.domain.com",
+						},
+						{
+							Domain: "hello.com",
+						},
+					}),
+				),
+			},
+		},
+	})
+}
+
+func getNetworkDef(s *terraform.State, name string) (*libvirtxml.Network, error) {
+	var network *libvirt.Network
+	rs, ok := s.RootModule().Resources[name]
+	if !ok {
+		return nil, fmt.Errorf("Not found: %s", name)
+	}
+	if rs.Primary.ID == "" {
+		return nil, fmt.Errorf("No libvirt network ID is set")
+	}
+	virConn := testAccProvider.Meta().(*Client).libvirt
+	network, err := virConn.LookupNetworkByUUIDString(rs.Primary.ID)
+	if err != nil {
+		return nil, err
+	}
+	networkDef, err := newDefNetworkfromLibvirt(network)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading libvirt network XML description: %s", err)
+	}
+	return &networkDef, nil
+}
+
+func testAccCheckLibvirtNetworkLocalOnlyStatus(name string, LocalOnlyStatus bool) resource.TestCheckFunc {
```

Comment:
`small nitpick` testAccCheckLibvirtNetworkLocalOnlyStatus should be called:


```ModuleTest_ + Testname``` 
``` TestAccCheckLibvirtNetwork_LocalOnlyStatus```

I know that this not documented so you coud not know it but we are trying to move all the testnames in this direction.

So when running Acceptance test we get more the debug context